File: frontend/src/types/room.ts
export interface Room {
    id: string;
    code: string;
    host_id: number;
    title: string;
    is_private: boolean;
    max_participants: number;
    current_movie_url: string | null;
    current_movie_title: string | null;
    current_position: number;
    is_playing: boolean;
    created_at: string;
    participants_count: number;
  }
  
  export interface RoomCreate {
    title: string;
    is_private: boolean;
    max_participants: number;
  }

File: services/rooms/app/schemas/room.py
from pydantic import BaseModel, Field
from uuid import UUID
from datetime import datetime
from typing import Optional


class RoomCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=100)
    is_private: bool = True
    max_participants: int = Field(default=10, ge=2, le=50)


class RoomResponse(BaseModel):
    id: UUID
    code: str
    host_id: int
    title: str
    is_private: bool
    max_participants: int
    current_movie_url: Optional[str] = None
    current_movie_title: Optional[str] = None
    current_position: int
    is_playing: bool
    created_at: datetime
    participants_count: int

    class Config:
        from_attributes = True


class JoinRoomResponse(BaseModel):
    room: RoomResponse
    user_role: str